src/main/groovy/com/deploygate/gradle/plugins/tasks/LoginTask.groovy

```diff
@@ -40,23 +69,23 @@ class LoginTask extends DefaultTask {
 
             logger.info("The authentication has succeeded. The application owner name is ${store.name}.")
 
-            // We can set the values iff it's not set yet because of the idempotency.
-
-            if (!deployGateExtension.appOwnerName) {
-                deployGateExtension.setAppOwnerName(store.name)
-            }
-
-            if (!deployGateExtension.apiToken) {
-                deployGateExtension.setApiToken(store.token)
-            }
+            // We can set the values unless they are found because of the idempotency.
+            credentials.appOwnerName.set(appOwnerName.getOrElse(store.name))
+            credentials.apiToken.set(apiToken.getOrElse(store.token))
+        } else {
+            credentials.appOwnerName.set(appOwnerName)
+            credentials.apiToken.set(apiToken)
         }
+
+        credentials.normalizeAndValidate()
```

Comment:
Before: Read variables from the extension and update them.
After: Receive variables as Inputs and set the result to Credentials.